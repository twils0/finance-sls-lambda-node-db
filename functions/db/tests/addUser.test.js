const { addUser } = require('../addUser');

global.pool = {
  query: jest.fn(),
};

const email = 'test1@test.com';
const emailAdditional = 'test2@test.com';
const userId = 'testUserID';
const emailPayload = {
  email,
};
const emailAdditionalPayload = {
  email,
  emailAdditional,
};
const dbResponse = {
  rows: [
    {
      user_id: userId,
    },
  ],
};

describe('functions', () => {
  describe('db', () => {
    describe('addUser', () => {
      afterEach(() => {
        global.pool.query.mockReset();
      });

      it('fails and returns unexpected error, when global.pool.query throws error, when provided email', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([email, null]);
          return Promise.reject(error);
        });

        try {
          await addUser(emailPayload);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('correctly adds email and returns userId, when provided email', async () => {
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([email, null]);
          return Promise.resolve(dbResponse);
        });

        const result = await addUser(emailPayload);

        expect(result).toEqual({ userId });
      });

      it('correctly adds email, emailAdditional, and returns userId, when provided email and emailAdditional', async () => {
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([email, emailAdditional]);
          return Promise.resolve(dbResponse);
        });

        const result = await addUser(emailAdditionalPayload);

        expect(result).toEqual({ userId });
      });
    });
  });
});
