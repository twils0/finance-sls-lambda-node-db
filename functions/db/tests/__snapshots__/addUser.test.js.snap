// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions db addUser correctly adds email and returns userId, when provided email 1`] = `
"INSERT INTO \\"user\\".\\"user\\" (email, email_additional)
      VALUES($1::text, $2::text) RETURNING user_id;"
`;

exports[`functions db addUser correctly adds email, emailAdditional, and returns userId, when provided email and emailAdditional 1`] = `
"INSERT INTO \\"user\\".\\"user\\" (email, email_additional)
      VALUES($1::text, $2::text) RETURNING user_id;"
`;

exports[`functions db addUser fails and returns unexpected error, when global.pool.query throws error, when provided email 1`] = `
"INSERT INTO \\"user\\".\\"user\\" (email, email_additional)
      VALUES($1::text, $2::text) RETURNING user_id;"
`;
